### PR TITLE
feat: add information on out-of-battle effects to move & ability info

### PR DESCRIPTION
**Pull request by mudkipscience**

### CODE REVIEW

**Review by Favna — CHANGES_REQUESTED** [review]
In abilities remove all the "from Pokemon emerald onward", "in geneartion X", etc. *Except* when it's exclusive to a generation(imem "only in Pokemon emerald" or "In Pokemon Sword & Shield").

Not even Bulbapedia is consistent and the inconsistency in this code is really annoying me. Take Snow Cloak for example and how it could've been listed as "From Generation IV onwards, ".

File: src/assets/moves.ts [hunk 1]
```diff
@@ -7855,7 +7867,8 @@ export default new GraphQLCollection<string, Pokemon.Move>([
       priority: 0,
       target: 'Self',
       type: 'Normal',
-      contestType: 'Cute'
+      contestType: 'Cute',
+      isFieldMove: "Tthe Pokémon cuts some of its HP to recover another Pokémon's HP. It can be learned by Miltank, Skiddo and Gogoat by leveling up."
```
Comment (Favna) [inline src/assets/moves.ts:7871]: Tthe -> The

File: src/assets/moves.ts [hunk 2]
```diff
@@ -4109,7 +4118,8 @@ export default new GraphQLCollection<string, Pokemon.Move>([
       target: 'Any',
       type: 'Flying',
       contestType: 'Clever',
-      maxMovePower: 130
+      maxMovePower: 130,
+      isFieldMove: 'The Pokemon can fly the player to any previously visited town/landmark. Fly can be taught to Pokemon with HM02 in all generations.'
```
Comment (Favna) [inline src/assets/moves.ts:4122]: "Pokemon" -> "Pokémon"

Also check if the same typo is elsewhere in items & abilities please. I'm pretty sure I did the rest a long while ago.

**Review by Favna — CHANGES_REQUESTED** [review]

**Comment by mudkipscience** [comment]
All done. Fixed some spelling errors, added the accent to Pokémon and removed the useless "from Pokémon Emerald onwards" text. I kept the "In Sword and Shield" stuff though, since currently those field effects only apply to those games.

**Comment by mudkipscience** [comment]
tests should be good now

**Review by Favna — APPROVED** [review]

**Pull request merged**